python_modules/libraries/dagster-dbt/dagster_dbt_tests/cli/test_app.py

```diff
@@ -47,6 +57,20 @@ def test_project_scaffold_command(monkeypatch: pytest.MonkeyPatch, tmp_project_d
     assert dagster_project_dir.joinpath(project_name).exists()
     assert not any(path.suffix == ".jinja" for path in dagster_project_dir.glob("**/*"))
 
+    subprocess.run(["dbt", "compile"], cwd=dbt_project_dir, check=True)
+
+    assert dbt_project_dir.joinpath("target", "manifest.json").exists()
+
+    monkeypatch.chdir(tmp_path)
+    sys.path.append(tmp_path.as_posix())
+
+    defs: "Definitions" = getattr(
```

Comment:
I'm surprised this type annotation would need quotes

Reply:
This is because `Definitions` is a conditional import since a `Definitions` object isn't being constructed and is only used for type hints.

Reply:
why does it need to be conditional though? I think it's fairly common to import something just to use for type hints, and the main reason for conditional is to avoid circular deps

btw this is non-blocking feedback

Reply:
Ah, my rule was that if the object isn't being used in the file, I usually enclose it in the `TYPE_CHECKING` guard, even if it isn't an expensive import.

I guess the loss in legibility isn't worth it. will change

Reply:
@sryza Funny enough, @smackesey is working on lint checks to automate this: https://github.com/dagster-io/dagster/pull/14675. Imports only used for type checks are now automatically under a `TYPE_CHECKING` conditional.